### PR TITLE
Port chardet to Python 3

### PR DESCRIPTION
**Pull request by PCManticore**
This PR includes changes needed to make `chardet==1.0.1` Python 3 compatible.

### CODE REVIEW

**Review by charettes — APPROVED** [review]
My main concern here is that this version doesn't have a test suite? It's really sad that we've diverged from upstream so much but at the same time I'm having trouble understanding why the project switched to BOM detection for UTF flavors.

You mentioned tests were passing locally on Slack. I assume you meant `zapier/zapier` test suite?

It would definitely be worth canarying this.

**Comment by PCManticore** [comment]
Yes, I was referring to `zapier/zapier` test suite. Definitely, I'll do a canary, but before that, I need to publish a version with these changes. Going to check if they added any tests with these commits, which were cherrypicked from https://github.com/chardet/chardet/releases/tag/2.2.0.

File: chardet/constants.py [hunk 1]
```diff
@@ -37,11 +38,3 @@
...
 
 SHORTCUT_THRESHOLD = 0.95
-
-import __builtin__
-if not hasattr(__builtin__, 'False'):
-    False = 0
-    True = 1
-else:
-    False = __builtin__.False
-    True = __builtin__.True
```
Comment (rsiemens) [inline chardet/constants.py:47]: This is strange. Wonder why this was something that was originally included...

Comment (charettes) [reply chardet/constants.py:47]: @rsiemens in reallllly old versions of Python there was no `True` or `False` builtin constants; the standard was to use `1` and `0`. That's the reason why `isinstance(True, int) and isinstance(False, int)` by the way.

Comment (rsiemens) [reply chardet/constants.py:47]: Ah TIL, thank you!

**Comment by PCManticore** [comment]
Alright, this is now ready. I had to cherry-pick a new commit from chardet 3.0.0, without which this version was detecting `Usér <to@gmail.com>` as ISO-8859-2 instead of UTF-8. The test suite seems to be passing locally for Python 3, I'll do a canary before bumping it.

**Pull request merged**